tests/storage/iso/conftest.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+import logging
+import time
+
+def setup_formatted_and_mounted_disk(host, sr_disk, fs_type, mountpoint):
+    device = '/dev/' + sr_disk
+    logging.info(f">> Format sr_disk {sr_disk} and mount it on host {host}")
+    host.ssh(['mkfs.' + fs_type, "-F", device])
+    host.ssh(['rm', '-rf', mountpoint]) # Remove any existing leftover to ensure rmdir will not fail in teardown
+    host.ssh(['mkdir', '-p', mountpoint])
+    host.ssh(['cp', '-f', '/etc/fstab', '/etc/fstab.orig'])
+    host.ssh(['echo', f'{device} {mountpoint} {fs_type} defaults 0 0', '>>/etc/fstab'])
+    host.ssh(['mount', mountpoint])
+
+def teardown_formatted_and_mounted_disk(host, mountpoint):
+    logging.info(f"<< Restore fstab and unmount {mountpoint} on host {host}")
+    host.ssh(['cp', '-f', '/etc/fstab.orig', '/etc/fstab'])
+    host.ssh(['umount', mountpoint])
+    host.ssh(['rmdir', mountpoint])
+
+@pytest.fixture(scope='session')
+def formatted_and_mounted_ext4_disk(host, sr_disk):
+    mountpoint = '/var/tmp/sr_disk_mountpoint'
+    setup_formatted_and_mounted_disk(host, sr_disk, 'ext4', mountpoint)
+    yield mountpoint
+    teardown_formatted_and_mounted_disk(host, mountpoint)
```

Comment:
À mon avis cette fixture aussi pourrait être utile dans d'autres tests, donc je la verrais bien dans le conftest.py global. En rendant le nom du mountpoint plus générique: `/var/tmp/local_disk_mountpoint` par exemple.

Reply:
if i move this fixture, is it in the conftest.py in the xcp-ng-tests (first place) ?
don't i need to move the `teardown_formatted_and_mounted_disk` function too ?

Reply:
Per one of my previous comments, you probably already moved those two functions into lib/common.py by now. In conftest.py you need to import them from there so that they are available in the fixture.

Reply:
i didn't move them to lib/common.py but in the conftest.py file.